src/mod/endpoints/mod_pulseaudio/pablio.c

```diff
@@ -0,0 +1,142 @@
+/*
+ * $Id: pablio.c 1151 2006-11-29 02:11:16Z leland_lucius $
+ * pablio.c
+ * PulseAudio abstraction
+ *
+ * Contributor(s):
+ *
+ * Phil Burk, http://www.softsynth.com
+ * Jérôme Poulin <jeromepoulin@gmail.com>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files
+ * (the "Software"), to deal in the Software without restriction,
+ * including without limitation the rights to use, copy, modify, merge,
+ * publish, distribute, sublicense, and/or sell copies of the Software,
+ * and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR
+ * ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include <switch.h>
+#include <stdlib.h>
+#include <string.h>
+#include <pulse/timeval.h>
+#include <pulse/simple.h>
+#include "pablio.h"
+
+long WriteAudioStream(PABLIO_Stream * aStream, void *data, size_t datalen, switch_timer_t *timer)
+{
+	pa_simple_write(aStream->ostream, data, datalen, NULL);
+	switch_core_timer_next(timer);
+
+	return datalen;
+}
+
+long ReadAudioStream(PABLIO_Stream * aStream, void *data, size_t datalen, switch_timer_t *timer)
+{
+	pa_simple_read(aStream->istream, data, datalen, NULL);
+	switch_core_timer_next(timer);
```

Comment:
```suggestion
```
Why use a timer? This function gets called inside the `channel_read_frame` callback, that is supposed synchronized.